### PR TITLE
[Feature] Enhanced FileSource to support more parameters feature

### PR DESCRIPTION
**Pull request by guoshupei**
### Related issues/PRs

Related issues: #4836 

### Checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [ ] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [Linkis mailing list](https://linkis.apache.org/community/how-to-subscribe) first)
- [ ] **If this is a code change**: I have written unit tests to fully verify the new behavior.

### CODE REVIEW

File: linkis-commons/linkis-storage/src/main/java/org/apache/linkis/storage/source/FileSource.java [hunk 1]
```diff
@@ -80,6 +80,14 @@ static boolean isResultSet(FsPath fsPath) {
     return isResultSet(fsPath.getPath());
   }
 
+  int getTotalCount();
```
Comment (peacewong) [inline linkis-commons/linkis-storage/src/main/java/org/apache/linkis/storage/source/FileSource.java:83]: The total number of lines is recommended to use getFileInfo

File: linkis-public-enhancements/linkis-script-dev/linkis-storage-script-dev-server/src/main/java/org/apache/linkis/filesystem/restful/api/FsRestfulApi.java [hunk 2]
```diff
@@ -583,12 +587,33 @@ public Message openFile(
...
+      }
       Pair<Object, List<String[]>> result = fileSource.collect()[0];
       IOUtils.closeQuietly(fileSource);
-      message.data("metadata", result.getFirst()).data("fileContent", result.getSecond());
+      message.data("metadata", result.getFirst());
+      message.data("fileContent", result.getSecond());
       message.data("type", fileSource.getFileSplits()[0].getType());
       message.data("totalLine", fileSource.getTotalLine());
-      return message.data("page", page).data("totalPage", 0);
+      message.data("totalCount", fileSource.getTotalCount());
```
Comment (peacewong) [inline linkis-public-enhancements/linkis-script-dev/linkis-storage-script-dev-server/src/main/java/org/apache/linkis/filesystem/restful/api/FsRestfulApi.java:609]: It is not recommended to return the total number of lines directly, it is recommended to call fileinfo：org.apache.linkis.filesystem.restful.api.FsRestfulApi#fileInfo

File: linkis-public-enhancements/linkis-script-dev/linkis-storage-script-dev-server/src/main/java/org/apache/linkis/filesystem/conf/WorkSpaceConfiguration.java [hunk 3]
```diff
@@ -61,6 +61,16 @@ public class WorkSpaceConfiguration {
   public static final CommonVars<Boolean> ENABLE_USER_GROUP =
       CommonVars$.MODULE$.apply("linkis.os.user.group.enabled", true);
 
+  public static final CommonVars<Integer> FILESYSTEM_LIMIT_TOTAL_LINE =
+      CommonVars$.MODULE$.apply("wds.linkis.filesystem.limit.total.line", 5000);
```
[the remark this answers is not in the document]
Comment (guoshupei) [reply linkis-public-enhancements/linkis-script-dev/linkis-storage-script-dev-server/src/main/java/org/apache/linkis/filesystem/conf/WorkSpaceConfiguration.java]: done

[the remark this answers is not in the document]
Comment (guoshupei) [reply linkis-public-enhancements/linkis-script-dev/linkis-storage-script-dev-server/src/main/java/org/apache/linkis/filesystem/conf/WorkSpaceConfiguration.java]: I'm going to remove this parameter

File: linkis-commons/linkis-storage/src/main/java/org/apache/linkis/storage/source/FileSource.java (hunk 1 above)
Comment (guoshupei) [reply linkis-commons/linkis-storage/src/main/java/org/apache/linkis/storage/source/FileSource.java]: Ok, good advice

File: linkis-public-enhancements/linkis-script-dev/linkis-storage-script-dev-server/src/main/java/org/apache/linkis/filesystem/restful/api/FsRestfulApi.java (hunk 2 above)
Comment (guoshupei) [reply linkis-public-enhancements/linkis-script-dev/linkis-storage-script-dev-server/src/main/java/org/apache/linkis/filesystem/restful/api/FsRestfulApi.java]: Ok, good advice

**Review by peacewong — APPROVED** [review]
LGTM.

**Pull request merged**